### PR TITLE
bump rust to nightly-2019-02-04

### PR DESCRIPTION
**Pull request by problame**

### CODE REVIEW

**Comment by flip1995** [comment]
`clippy::new_without_default_derive` got merged into `new_without_default`. https://travis-ci.org/comprakt/comprakt/jobs/488696895#L927-L933

File: parser/src/parser.rs [hunk 1]
```diff
@@ -630,18 +629,15 @@ where
     /// [1]: https://en.wikipedia.org/wiki/Shunting-yard_algorithm
     /// [2]: https://en.wikipedia.org/wiki/Reverse_Polish_notation#Postfix_evaluation_algorithm
     /// [3]: https://eli.thegreenplace.net/2012/08/02/parsing-expressions-by-precedence-climbing
-    fn parse_binary_expression(&mut self) -> BoxedResult<'f, ast::Expr<'f>> {
+    fn parse_binary_expression(&mut self) -> ParserResult<'f, ast::Expr<'f>> {
         let mut operator_stack = Vec::new();
         let mut operand_stack = Vec::new();
-        fn rpn_eval<'f>(
-            operand_stack: &mut Vec<Box<Spanned<'f, ast::Expr<'f>>>>,
```
Comment (flip1995) [inline parser/src/parser.rs]: Clippy wanted me to remove this `Box`. So I did. This led to a huge fallout. But by the time I realized how big the fallout was, I was in too deep already.

**Comment by reiner-dolp** [comment]
isn't this unnecessarily risky?

File: lexer/src/lexer.rs [hunk 2]
```diff
@@ -27,7 +27,6 @@ macro_rules! match_op {
     ($input:expr, $span:ident, $len:expr, $right:expr) => {{
         // Unwraps are safe, because this is only called after token of length $len,
         // is already matched and thus contained in $input
-        debug_assert!($len >= 1);
```
Comment (flip1995) [inline lexer/src/lexer.rs]: Maybe wait on rust-lang/rust-clippy#3740 or allow the clippy lint here instead of removing it.

**Pull request closed without merging**